python/cudf_polars/cudf_polars/experimental/utils.py

```diff
@@ -55,9 +55,7 @@ def _lower_ir_fallback(
     if fallback and msg:
         # Warn/raise the user if any children were collapsed
         # and the "fallback_mode" configuration is not "silent"
-        match fallback_mode := rec.state["config_options"].get(
-            "executor_options.fallback_mode"
-        ):
+        match fallback_mode := rec.state["config_options"].executor.fallback_mode:
```

Comment:
I missed this in the refactor earlier, because `rec.state` is just typed as `Mapping[str, Any]` so mypy didn't see that `rec.state["config_options"].get` would be an `AttributeError`.

If we think the set of things going into `rec.state` is pretty stable, we might consider making it a `typing.TypedDict`. If it's pretty dynamic, then we'll just need to be careful.